File: backend/src/main/java/es/codeurjc/exercise4you/entity/questionnaire/QuestionnaireInfo.java
package es.codeurjc.exercise4you.entity.questionnaire;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class QuestionnaireInfo {
    private boolean alreadyExists;
    private Question question;
    private List<Alert> alertList;

    @Data
    @AllArgsConstructor
    public static class Alert {
        private String title;
        private String alert;
    }
}

File: frontend/src/app/model/questionnaires/questionnaire-info.ts
export interface QuestionnaireInfo {
    alreadyExists:boolean,
    question: {
        id: string,
        code: string,
        type: string,
        description: string,
        introduction: string,
        question: string,
        options: string[],
    },
    alertList: {
        title: string,
        alert: string,
    }[],
}